### PR TITLE
tests: lwm2m_registry: Add ZTest tests for lwm2m_registry

### PR DESCRIPTION
**Pull request by jarlamsa**
These tests test the basic functionality of creating object and resource instances, setting buffers, writing and reading to and from resources, and setting and triggering callbacks.

Signed-off-by: Jarno Lämsä <jarno.lamsa@nordicsemi.no>

### CODE REVIEW

**Comment by jarlamsa** [comment]
@SeppoTakalo @rlubos @carlescufi please review.

**Comment by jarlamsa** [comment]
@juhaylinen please review.

**Review by juhaylinen — DISMISSED** [review]

File: tests/subsys/net/lib/lwm2m/lwm2m_registry/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+target_include_directories(app
+  PRIVATE
+  ${includes}
+)
+
+add_compile_definitions(CONFIG_LWM2M_IPSO_TEMP_SENSOR=y)
+add_compile_definitions(CONFIG_LWM2M_IPSO_TEMP_SENSOR_VERSION_1_1=y)
+add_compile_definitions(CONFIG_LWM2M_IPSO_TEMP_SENSOR_INSTANCE_COUNT=1)
```
Comment (rlubos) [inline tests/subsys/net/lib/lwm2m/lwm2m_registry/CMakeLists.txt:30]: Ain't this redundant with the project file? Those are already enabled via Kconfig.

File: tests/subsys/net/lib/lwm2m/lwm2m_registry/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(lwm2m_client_unittest)
+
+FILE(GLOB app_sources src/*.c)
+target_sources(app
+  PRIVATE
+    ${app_sources}
+    $ENV{ZEPHYR_BASE}/subsys/net/lib/lwm2m/ipso_temp_sensor.c
```
Comment (rlubos) [inline tests/subsys/net/lib/lwm2m/lwm2m_registry/CMakeLists.txt:15]: Ditto, this should be pulled in by the build system (CONFIG_LWM2M_IPSO_TEMP_SENSOR is enabled)

File: tests/subsys/net/lib/lwm2m/lwm2m_registry/testcase.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+common:
+  depends_on: netif
+tests:
+  net.lwm2m.engine:
+    platform_allow: native_posix
```
Comment (rlubos) [inline tests/subsys/net/lib/lwm2m/lwm2m_registry/testcase.yaml:5]: Do we need to limit tests to `native_posix` only? Would be good to allow them to run on other platforms as well.

**Comment by jarlamsa** [comment]
Thank you @rlubos for your review. I think have addressed all of your review comments. Please re-review.

**Review by SeppoTakalo — CHANGES_REQUESTED** [review]
Please move to `tests/net/lib` where rest of the LwM2M tests are

I'll move the RD client test too in https://github.com/zephyrproject-rtos/zephyr/pull/54158

File: tests/subsys/net/lib/lwm2m/lwm2m_registry/src/lwm2m_registry.c [hunk 4]
```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/fff.h>
```
Comment (SeppoTakalo) [inline tests/subsys/net/lib/lwm2m/lwm2m_registry/src/lwm2m_registry.c:8]: I don't see any stubs, is the `fff.h` unneeded here?

Comment (jarlamsa) [reply tests/subsys/net/lib/lwm2m/lwm2m_registry/src/lwm2m_registry.c:8]: Oh, that's true.

**Comment by jarlamsa** [comment]
Thank you @SeppoTakalo for your review comments, those have now been addressed, please re-review.

**Review by rlubos — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

File: tests/net/lib/lwm2m/lwm2m_registry/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Comment (stephanosio) [inline tests/net/lib/lwm2m/lwm2m_registry/CMakeLists.txt:4]: This is not allowed upstream.

Comment (SeppoTakalo) [reply tests/net/lib/lwm2m/lwm2m_registry/CMakeLists.txt:4]: Good catch. We'll fix.

**Pull request merged**